src/common/msb.rs

```diff
@@ -1,26 +1,36 @@
+#[derive(Debug, Eq, PartialEq)]
+pub enum Bit {
```

Comment:
What is the main reasoning behind breaking out `Bit` into it's own type and not just using `u8`? Also was there any reason to not just use [associated-constants](https://doc.rust-lang.org/reference/items/associated-items.html#associated-constants) and make `0` and `1` there own specific variants?